plugins/user-settings/src/components/AuthProviders/ProviderSettingsItem.tsx

```diff
@@ -45,17 +46,21 @@ export const ProviderSettingsItem = ({
 }: Props) => {
   const api = useApi(apiRef);
   const [signedIn, setSignedIn] = useState(false);
+  const [profileEmail, setProfileEmail] = useState('');
 
   useEffect(() => {
     let didCancel = false;
 
-    const subscription = api
-      .sessionState$()
-      .subscribe((sessionState: SessionState) => {
-        if (!didCancel) {
-          setSignedIn(sessionState === SessionState.SignedIn);
-        }
-      });
+    const subscription = api.sessionState$().subscribe(() => {
+      if (!didCancel) {
+        api
+          .getProfile({ optional: false })
+          .then((profile: ProfileInfo | undefined) => {
+            setSignedIn(profile !== undefined);
```

Comment:
You should probably still check the sessionState out of the observable for this. If that's signed in, we should probably set this state to true no matter what.

Reply:
okay, is there any reason that makes it a good thing? purely from personal interest of why it would be better